src/SupabaseClient.ts

```diff
@@ -329,13 +329,13 @@ export default class SupabaseClient<
     source: 'CLIENT' | 'STORAGE',
     token?: string
   ) {
+    this.realtime.setAuth()
```

Comment:
**question**

Not a lot of context here, but I wonder why we call it on top of the function an not instead something like:

```ts
_handleTokenChanged() {
   ...
   this.realtime.setAuth(this.changedAccessToken)
}
```

Reply:
We changed it to be a callback function so on token change realtime runs the `accessToken` callback and pulls credentials as defined by said callback and pushes it to connected channels (required for proper Realtime operations)

https://github.com/supabase/realtime-js/blob/9ef764907abc202ce0efd5769149660b4e0f928c/src/RealtimeClient.ts#L357

Reply:
Here's the code on the supabase-js side https://github.com/supabase/supabase-js/blob/325c2c9b253a81cf1f02ec7e2bdbced7318467f8/src/SupabaseClient.ts#L122